File: src/fixtures/goproject/base_test.go
package fixture

import "testing"

// Base tests: covers Add, Divide, Reverse, PadLeft
// Does NOT cover: Subtract, Multiply, Modulo, Capitalize, Truncate

func TestAddBase(t *testing.T) {
	if Add(1, 2) != 3 {
		t.Fatal("expected 3")
	}
}

func TestDivideBase(t *testing.T) {
	result, err := Divide(10, 2)
	if err != nil {
		t.Fatal(err)
	}
	if result != 5 {
		t.Fatal("expected 5")
	}
}

func TestReverseBase(t *testing.T) {
	if Reverse("abc") != "cba" {
		t.Fatal("expected cba")
	}
}

func TestPadLeftBase(t *testing.T) {
	if PadLeft("hi", 5, '*') != "***hi" {
		t.Fatal("expected ***hi")
	}
}

File: src/fixtures/bun/base.test.ts
import {
	expect,
	test,
} from "bun:test";

import {
	add,
	divide,
} from "./calc";
import {
	padLeft,
	reverse,
} from "./strutil";

// Base tests: covers add, divide, reverse, padLeft
// Does NOT cover: subtract, multiply, modulo, capitalize, truncate

test("add", () => {
	expect(add(1, 2)).toBe(3);
});

test("divide", () => {
	expect(divide(10, 2)).toBe(5);
	expect(() => divide(1, 0)).toThrow("division by zero");
});

test("reverse", () => {
	expect(reverse("abc")).toBe("cba");
});

test("padLeft", () => {
	expect(padLeft("hi", 5, "0")).toBe("000hi");
	expect(padLeft("hello", 3, "0")).toBe("hello");
});
